### DISCUSSION TITLE
I encountered problems using Vite and want to know if it is available here?

### DISCUSSION THREAD

**Discussion by Mauladen** [farm-fe/farm#1472] [General]
Hello everyone, I would like to know if Farm has the same problems?

1. [Vite does not use treeshaking in dev mode](https://github.com/vitejs/vite/issues/8237)
2. [Vite breaks the CSS order](https://github.com/vitejs/vite/issues/3924)

**Comment by wre232114** [comment]
NO, Farm does not have the same problems:
1. One of Farm design philosophy is `consistency`. So you can enable tree shake and minify in dev mode
2. Farm respects the dependency order of css.

**Reply by Mauladen** [reply]
> NO, Farm does not have the same problems:
> 
>     1. One of Farm design philosophy is `consistency`. So you can enable tree shake and minify in dev mode
> 
>     2. Farm respects the dependency order of css.

Thank you for your answer, but I did not find information on how to enable true shake in development mode, according to the documentation, the [server](https://www.farmfe.org/docs/config/dev-server) does not have such a setting

**Reply by wre232114** [reply]
Just set `compilation.treeShake: true`. Tree shake would work in both dev and prod.

**Reply by wre232114** [reply]
All options under [compilation](https://www.farmfe.org/docs/config/compilation-options#compilation-options---compilation) work for dev and prod

**Reply by Mauladen** [reply]
Thank you, it's very cool
Have a nice day
